addons/web_editor/static/src/scss/web_editor.common.scss

```diff
@@ -807,6 +807,13 @@ blockquote {
                     border-color: $-btn-primary-color;
                 }
             }
+
+            // Forms
+            @if ($-bg-color == $light and $-bg-color != $body-bg) {
+                --o-input-bg: #{$body-bg};
+                --o-input-border-color: var(--o-border-color);
+                --o-input-placeholder-color: #{rgba(color-contrast($body-bg), .5)};
+            }
```

Comment:
> Maybe some more magic [...]

I see what you mean, but this is not exactly "magic" since it address a very specific scenario.
The condition could be simplified to
```scss
@if ($index == 2) {}
```
The reason why it's coded this way is to keep the "door open" for advanced customization, eg. if the user wants to force `$light` to be something different than `color-2`    

> ... + the ability to override for each preset?

This would be a very welcome :) I think @pierre-pde already agreed, we didn't implement it because it's quite complex and honestly out-of-scope considering that what we have in this  PR works.

Reply:
Conclusion of IRL talk: not for now, light inputs except in light-colored background is fine for now (and still way better than current master IMO)

Reply:
Not sure the `and $-bg-color != $body-bg` in the condition is what you actually wanted:

![image](https://github.com/odoo/odoo/assets/10338094/45321720-b663-42d0-ae13-aefcf46d9859)

I'll keep it for now as it is, we can make a fix if needed but $body-bg == $light is a very specific case so not awful if kept as it is.